src/SpendPermissionManager.sol

```diff
@@ -375,65 +434,95 @@ contract SpendPermissionManager is EIP712 {
         _useSpendPermission(spendPermission, value);
         _execute({
             account: spendPermission.account,
-            target: magicSpend,
+            target: MAGIC_SPEND,
             value: 0,
             data: abi.encodeWithSelector(MagicSpend.withdraw.selector, withdrawRequest)
         });
         _transferFrom(spendPermission.token, spendPermission.account, spendPermission.spender, value);
     }
 
-    /// @notice Approves a permission while revoking another if its last update has not changed.
+    /// @notice Check if a spend permission is revoked.
     ///
-    /// @dev Enforces that the last updated period of the permission being revoked matches the last valid updated period
-    ///      submitted as an argument. This is to prevent frontrunning `approveWithRevoke` with additional last-minute
-    ///      spends.
-    /// @dev The `account` of the permissions must match, but the remaining fields can differ.
-    /// @dev Can only be called by the `account` of a permission.
+    /// @param spendPermission Details of the spend permission.
     ///
-    /// @param permissionToApprove Details of the spend permission to approve.
-    /// @param permissionToRevoke Details of the spend permission to revoke.
-    /// @param lastValidUpdatedPeriod Last valid updated period for the spend permission being revoked.
+    /// @return revoked True if spend permission is revoked.
+    function isRevoked(SpendPermission memory spendPermission) public view returns (bool) {
```

Comment:
nit: no internal usage, so keep visibility tight
```suggestion
    function isRevoked(SpendPermission memory spendPermission) external view returns (bool) {
```

Might be worth keeping as `public` though in case anyone inheriting would want to use this 🤷 ?

Reply:
I'm tempted to leave this public bc it leaves open that possibility and there's no cost to how they're intended to be used, as getters